src/Microsoft.OData.Core/Metadata/EdmLibraryExtensions.cs

```diff
@@ -448,11 +445,11 @@ internal static string FullNameWithParameters(this IEdmOperationImport operation
         /// <param name="source">The source.</param>
         /// <param name="actionItems">The action items.</param>
         /// <returns>Only the functions from the operation sequence.</returns>
-        internal static IEnumerable<IEdmFunction> RemoveActions(this IEnumerable<IEdmOperation> source, out IList<IEdmAction> actionItems)
+        internal static IEnumerable<IEdmOperation> RemoveActions(this IEnumerable<IEdmOperation> source, out IList<IEdmOperation> actionItems)

```

Comment:
>IEnumerable<IEdmOperation> [](start = 24, length = 26)

Why is the return type changed to IEdmOperation? the method still only returns the functions in this list, and actions as an output param.

Reply:
Because the return will be used to call "FilterOperationsByParameterNames", which accepts the "IEdmOperation". We don't want to cast the operation to function, then cast the function to operation again and again. That's useless and waste of time. 



---
In reply to: [289995838](https://github.com/OData/odata.net/pull/1473#discussion_r289995838) [](ancestors = 289995838)